File: io/json/src/test/resources/compat/issue-754/src/java/LegacyJsonImplementation.java
package io.bluetape4k.json.compat.issue754.java;

import io.bluetape4k.json.JsonSerializer;
import java.nio.charset.StandardCharsets;

public final class LegacyJsonImplementation implements JsonSerializer {
    public int serializeCalls;
    public int deserializeCalls;

    @Override
    public byte[] serialize(Object graph) {
        serializeCalls++;
        return graph == null ? new byte[0] : graph.toString().getBytes(StandardCharsets.UTF_8);
    }

    @Override
    @SuppressWarnings("unchecked")
    public <T> T deserialize(byte[] bytes, Class<T> clazz) {
        deserializeCalls++;
        Object value = bytes == null || bytes.length == 0
            ? null
            : new String(bytes, StandardCharsets.UTF_8);
        return value == null ? null : (T) value;
    }
}

File: io/json/src/test/resources/compat/issue-754/src/kotlin/LegacyJsonImplementation.kt
package io.bluetape4k.json.compat.issue754.kotlin

import io.bluetape4k.json.JsonSerializer

class LegacyJsonImplementation: JsonSerializer {
    override fun serialize(graph: Any?): ByteArray = graph?.toString()?.encodeToByteArray() ?: byteArrayOf()

    @Suppress("UNCHECKED_CAST")
    override fun <T: Any> deserialize(bytes: ByteArray?, clazz: Class<T>): T? =
        bytes?.takeIf { it.isNotEmpty() }?.decodeToString() as T?
}
